Source/SwiftLintFramework/Rules/Lint/UnusedImportRule.swift

```diff
@@ -225,13 +225,10 @@ private extension SwiftLintFile {
 
     func rangedAndSortedUnusedImports(of unusedImports: [String], contents: NSString) -> [(String, NSRange)] {
         return unusedImports
-            .map { module in
-                let testableImportRange = contents.range(of: "@testable import \(module)\n")
-                if testableImportRange.location != NSNotFound {
-                    return (module, testableImportRange)
+            .compactMap { module in
+                return self.match(pattern: #"^(@testable )?import \#(module)\b.*?\n"#).first.map { match in
```

Comment:
This will silently swallow modules in `unusedImports` that fail to match the regex. Should this be a `guard` with a warning in the unmatched case?

Reply:
There's an unlikely case that could cause problems:

```swift
import Foo /* I hope no-one does this
because it would be bad */
```

I wonder if SwiftLint should be conservative and remove just the import, and leave anything that follows in place.

Reply:
```suggestion
                return self.match(pattern: #"^(@testable\s+)?import\s+\#(module)\b.*?\n"#).first.map { match in
```

May want to change the literal spaces to `\s+` to ensure that multiple spaces and tabs are matched too.

Reply:
> This will silently swallow modules in `unusedImports` that fail to match the regex. Should this be a `guard` with a warning in the unmatched case?

There shouldn't be any inputs here that fail the regex. If there are it's a bug in SwiftLint, not user error, so I'd rather fail loudly.

---

The multiline comment case isn't something I care to handle.

---

I'll apply the `\s+` changes, good idea!

Reply:
> There shouldn't be any inputs here that fail the regex. If there are it's a bug in SwiftLint, not user error, so I'd rather fail loudly.

I think we're agreeing. But right now, this code will be silent in the case the pattern doesn't match. For example `@_exported import Foo` will silently be dropped https://github.com/realm/SwiftLint/issues/2877.

Reply:
Ah thanks, I didn't know about that issue. I've added support for that now in 45fa67c.